### ISSUE TITLE
Cannot set SOCKS proxy in GeckoDriver

### ISSUE THREAD

**Issue by AlexeiScherbakov** [SeleniumHQ/selenium#6938]
## 🐛 Bug Report

Cannot set socks proxy type in FirefoxOptions

throws `'InvalidArgumentError: Expected "socksVersion" to be a positive integer, got [object Undefined] undefined (SessionNotCreated)'`

## To Reproduce
Use FirefoxOptions with Socks proxy configured. For example

```
FirefoxOptions options = new FirefoxOptions
	{
		Profile = profile,
		PageLoadStrategy = PageLoadStrategy.Eager,
		BrowserExecutableLocation = browserExecutable,
		Proxy = new Proxy()
		{
			Kind = ProxyKind.Manual,
			HttpProxy = "127.0.0.1:8118",
			SslProxy = "127.0.0.1:8118",
			SocksProxy = "127.0.0.1:1080",
		},
	};
```

## Expected behavior

I think that field with SocksProxy version must exists in Proxy class

## Test script or set of commands reproducing this issue

```
	var profile="D:\\test-profile";
	var browserExecutable="C:\\Program Files\\Mozilla Firefox";
	FirefoxOptions options = new FirefoxOptions
	{
		Profile = profile,
		PageLoadStrategy = PageLoadStrategy.Eager,
		BrowserExecutableLocation = browserExecutable,
		Proxy = new Proxy()
		{
			Kind = ProxyKind.Manual,
			HttpProxy = "127.0.0.1:8118",
			SslProxy = "127.0.0.1:8118",
			SocksProxy = "127.0.0.1:1080",
		},
	};
	_driver = new FirefoxDriver(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), options,System.Threading.Timeout.InfiniteTimeSpan);
```

## Environment

OS: Windows 10
Browser: Firefox
Browser version: Firefox 64
Browser Driver version: geckodriver-v0.24.0
Language Bindings version: C# 3.141.0

**Comment by p0deje** [comment]
Thank you for the bug report! It looks like Java/Node.js are the only clients that support `socksVersion`, so this needs to be fixed everywhere else. Here is the link to relevant part of spec: https://w3c.github.io/webdriver/#proxy.

**Comment by twalpole** [comment]
Fixed in ruby via 6a717f202bc8fced4726d980e9657d305960896b

**Comment by jimevans** [comment]
Added to .NET in 6b8bbd3

**Comment by singh811** [comment]
Fixed in Python in this #6983

**Comment by lmtierney** [comment]
Closing this as python was added with #6983
